plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -518,10 +560,23 @@ public JSONObject getJson(String urlString) {
      */
     public void getUCJson() {
         logVerbose("\nRetrieving update center information");
-        latestUcJson = getJson(jenkinsUcLatest);
+        createCacheManager();
+        cm.createCache();
+
+        latestUcJson = getJson(stringToUrlQuietly(jenkinsUcLatest), "update-center");
         latestPlugins = latestUcJson.getJSONObject("plugins");
-        experimentalUcJson = getJson(cfg.getJenkinsUcExperimental().toString());
-        pluginInfoJson = getJson(Settings.DEFAULT_PLUGIN_INFO_LOCATION);
+        experimentalUcJson = getJson(cfg.getJenkinsUcExperimental(), "experimental-update-center");
+        pluginInfoJson = getJson(stringToUrlQuietly(Settings.DEFAULT_PLUGIN_INFO_LOCATION), "plugin-versions");
+    }
+
+    private void createCacheManager() {
+        Path path = jenkinsWarFile.toPath();
+        Path parent = path.getParent();
+        if (parent == null) {
+            throw new IllegalArgumentException("Jenkins WAR file has no parent directory");
+        }
+        Path cachePath = parent.resolve("plugin-installation-tool-cache");
+        cm = new CacheManager(cachePath, verbose);
```

Comment:
This dies in the tests for the docker images because it tries to write to a directory that the user doesn't have access to.

```
java.io.UncheckedIOException: java.nio.file.AccessDeniedException: /usr/share/jenkins/plugin-installation-tool-cache
        at io.jenkins.tools.pluginmanager.impl.CacheManager.createCache(CacheManager.java:31)
        at io.jenkins.tools.pluginmanager.impl.PluginManager.getUCJson(PluginManager.java:564)
        at io.jenkins.tools.pluginmanager.impl.PluginManager.start(PluginManager.java:125)
        at io.jenkins.tools.pluginmanager.cli.Main.main(Main.java:37)
     Caused by: java.nio.file.AccessDeniedException: /usr/share/jenkins/plugin-installation-tool-cache
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:384)
        at java.nio.file.Files.createDirectory(Files.java:674)
        at io.jenkins.tools.pluginmanager.impl.CacheManager.createCache(CacheManager.java:26)
        ... 3 more
     java.nio.file.AccessDeniedException: /usr/share/jenkins/plugin-installation-tool-cache
```